// SPDX-FileCopyrightText: Copyright (c) 2025 Cisco and/or its affiliates.
// SPDX-License-Identifier: Apache-2.0

syntax = "proto3";

package core.v1alpha1;

// Reference to the object in the content store.
message ObjectRef {
  // CID of the object
  // Specs: https://github.com/multiformats/cid
  string cid = 1;
  // Name of the object. Can be resolved from CID.
  string name = 2;
  // Kind of the object. Can be resolved from CID.
  ObjectKind kind = 3;
  // Size of the object. Can be resolved from CID.
  uint64 size = 4;
}

// The actual object stored in the content store.
message Object {
  // Name of the object
  string name = 1;
  // Kind of the object
  ObjectKind kind = 2;
  // Size of the object
  uint64 size = 3;
  // Opaque data held by this object. Arbitrary in size.
  // Readers should know how to process this value.
  bytes data = 4;
}

// ObjectKind specifies supported objects by kind.
enum ObjectKind {
  OBJECT_TYPE_UNSPECIFIED = 0;
  OBJECT_TYPE_RAW = 1;
  OBJECT_TYPE_AGENT = 2;
  OBJECT_TYPE_EXTENSION = 3;
}
